yt_dlp/extractor/nebula.py

```diff
@@ -73,10 +73,23 @@ def _retrieve_nebula_api_token(self):
     def _call_nebula_api(self, url, video_id=None, method='GET', auth_type='api', note=''):
         assert method in ('GET', 'POST',)
         assert auth_type in ('api', 'bearer',)
-        authorization = f'Token {self._nebula_api_token}' if auth_type == 'api' else f'Bearer {self._nebula_bearer_token}'
-        return self._download_json(
-            url, video_id, note=note, headers={'Authorization': authorization},
-            data=b'' if method == 'POST' else None)
+
+        def inner_call():
+            authorization = f'Token {self._nebula_api_token}' if auth_type == 'api' else f'Bearer {self._nebula_bearer_token}'
+            return self._download_json(
+                url, video_id, note=note, headers={'Authorization': authorization},
+                data=b'' if method == 'POST' else None)
+
+        try:
+            return inner_call()
+        except ExtractorError as exc:
+            # if 401 or 403, attempt credential re-auth and retry
+            if exc.cause and isinstance(exc.cause, urllib.error.HTTPError) and exc.cause.code in (401, 403):
+                self.to_screen(f'Reauthenticating to Nebula and retrying, because last {auth_type} call resulted in error {exc.cause.code}')
+                self._real_initialize()
```

Comment:
perhaps move the code to a `_login` function and then just call it from both _real_initialize and here. While it is effectively same, it should make the intend clearer (and also makes it simpler to add other things to _real_initialize if needed)

Reply:
I agree, makes sense.